### DISCUSSION TITLE
Consulta sobre línea de comandos, mensajes y criterios de seleccion de proveedores.

### DISCUSSION THREAD

**Discussion by Gruschka** [dds-jv/foro#142] [TPA - 2020] [answered]
Buenas!

Tenemos algunas dudas sobre la funcionalidad que tiene que tener la linea de comandos para esta entrega y sobre los criterios de selección de proveedores:

**Respecto a la linea de comandos**: 

El enunciado dice que se tiene que poder ejecutar de forma manual la validacion de egresos. Ahora bien, es sólo esto lo que se tiene que poder hacer? Es decir:

- Los egresos que estén pendientes de validación se tienen que poder cargar también por consola o los podríamos crear en el mismo main (medio como un test) para que alguien pueda correr la validación?

- Los mensajes con los resultados de las validaciones tienen que mostrarse? Si es así, necesitaríamos una forma de discriminar inequívocamente un egreso de otro. Podemos hacer una forma arbitraria de identificación de egresos? 

**Respecto a los mensajes con los resultados de validación**: 

- Tiene que decir si un Egreso X pasó o falló la validación o si la fallo tiene que especificar qué validación no pudo superar?

**Respecto a la selección de proveedores**: 

El TP dice lo siguiente: 

`Si el criterio de selección del proveedor fuera el de menor valor, debe validarse que, entre los presupuestos cargados en el sistema, se haya seleccionado el de menor valor. 
`
- El criterio de selección, es a nivel Egreso, Entidad u Organización? Es decir, puedo tener dentro de una misma entidad egresos que se seleccionen por menor valor y otros que no?

- En caso de que no se este usando dicho criterio, directamente en las validaciones no tendríamos que contemplar esta ultima y ya?

Muchas gracias chicos!

**Comment by gastonprieto** [comment, marked as answer]
Respecto a la linea de comandos:

* No es necesaria crear una interfaz gráfica para darlo de alta, pueden asumir que ya estan creados.
* No es necesario que lo muestren sino que tengan el efecto necesario, para saber si andan tienen los tests, pero nadie va a estar viendo el resultado de su proceso. A lo sumo pueden investigar lo que es logging y mostrar algunos mensajes por si alguien quiere ver los logs.

Respecto a los mensajes con los resultados de validación:

* Analicen ambas alternativas y discutanlas con su ayudante, pero piensen cual es el trade off de implementar una u otra solución y cual serviría mas al usuario final.

> Respecto a la selección de proveedores:

* Por ahora es a nivel organización no dicen de otra cosa, incluso tampoco estamos muy seguro de los otros criterios que puedan haber.

* Acá creo que depende de como lo modeles, pero si no lo estas usando significa que no aplica.

**Comment by roli93** [comment]
@Gruschka Les sirvió?

**Comment by Gruschka** [comment]
> @Gruschka Les sirvió?

yessir.

Muchisimas gracias @gastonprieto  y perdón por no cerrar el issue, se me super pasó.
